awx/api/serializers.py

```diff
@@ -2206,8 +2212,24 @@ def get_related(self, obj):
         if self.version > 1:
             res['credentials'] = self.reverse('api:inventory_update_credentials_list', kwargs={'pk': obj.pk})
 
+        source_project = self.get_source_project(obj)
+        if source_project:
+            res['source_project'] = self.reverse('api:project_detail', kwargs={'pk': source_project})
+
         return res
 
+    def get_summary_fields(self, obj):
+        summary_fields = super(UnifiedJobSerializer, self).get_summary_fields(obj)
+        source_project = getattrd(obj, 'source_project_update.unified_job_template', None)
```

Comment:
seem like you would want `source_project = self.get_source_project(obj)` here

Reply:
this might be more to do with bad variable naming on my part - we want only the id to show up in the toplevel api data, so we return just the id with the `source_project` MethodField. 

When generating the summary fields we want the object data. I'm wondering if this variable would be better named as `source_project_obj` or something.

Reply:
My preference would be that `get_source_project` returns the object and then if you need the id you do `get_source_project(obj).pk`. You can also do `getattrd(obj, 'source_project_update.unified_job_template_id', None)`, which sometimes has a performance benefit, but not in this case because you wind up fetching the object one way or the other.

Reply:
alright - `get_source_project` now returns the object